app/controllers/users/sessions_controller.rb

```diff
@@ -51,6 +55,34 @@ def login_token_expired?
     Time.zone.now > @user.login_token_valid_until
   end
 
+  def user_requires_magic_link(user)
+    return false unless user&.admin?
+
+    environment_allows_test_users = Rails.env.development? || Rails.env.deployed_development? || Rails.env.staging?
+    user_is_test_user = TEST_USERS.include?(user.email)
+    return false if environment_allows_test_users && user_is_test_user
+
+    true
+  end
+
+  def send_magic_link(user)
+    token_expiry = 60.minutes.from_now
+    result = user&.update(
+      login_token: SecureRandom.hex(10),
+      login_token_valid_until: token_expiry,
+    )
+
+    if result
+      url = Rails.application.routes.url_helpers.users_confirm_sign_in_url(
+        login_token: user.login_token,
+        host: Rails.application.config.domain,
+        **UtmService.email(:sign_in),
+      )
+
+      UserMailer.sign_in_email(user: user, url: url, token_expiry: token_expiry.localtime.to_s(:time)).deliver_now
```

Comment:
Why did the mailer need requiring in for the old strategy but not here?

Reply:
Rails magic :D